model/user/user.go

```diff
@@ -65,6 +66,26 @@ func (u *DBUser) GetPublicKey(keyname string) (string, error) {
 	return "", errors.Errorf("Unable to find public key '%v' for user '%v'", keyname, u.Username())
 }
 
+func (u *DBUser) AddPublicKey(keyName, keyValue string) error {
```

Comment:
I seem to have a memory that DBUser is often accessed via an interface? so adding a method to this might be odd in some cases, but I think its fine.

Reply:
This implementation already has a bunch of other functions specific to it, so I don't think this is a big deal